### ISSUE TITLE
Markdown: do not remove leading & trailing no-break whitespaces in inline code blocks

### ISSUE THREAD

**Issue by maximbaz** [prettier/prettier#3755]
**Prettier 1.10.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAHATnAzndA3OAAgBs4BDAEwEsoBzQsqCwmdMq4m+qCAWgCNMZANaEA7gAsq8LKjJhshGkqicoRSBThJCAAwAFhiXGLEIh-bpAAaEBFQwq0LMlBl06CGIAK7hC5QyPAgqChsQMiwYZAAzMmIcW0F5YTgYAGU5MC5kVgBXOFsKCDBY+MSQGhx0GG82WgBbMjKEwpAAKywADwAhNjBUjLIGuAAZGjgWiog8mFRZgCYptrl0auQQJvRhYrEocIwaGAB1UJgJZAAOAAZbDAgcY7ZUDYxsXAJwzABHPKpMOpkRrNJBxVq2HANKi5dAFCFcUgARTyEHgy1sMDI-FOFHOyAWGLYHC4AGEIA0mhseOpwnkcAAVLEBME4AC+rKAA)
```sh
--parser markdown
```

**Input:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: `   hello   `
```

**Output:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: `hello`

```

**Expected behavior:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: `   hello   `
```

**Comment by maximbaz** [comment]
Reported https://github.com/remarkjs/remark/issues/321

**Comment by shreyasminocha** [comment]
It seems that remarkjs/remark#321 was closed with remarkjs/remark#424. However, I can still reproduce this issue with Prettier 2.0.0.

**Comment by sosukesuzuki** [comment]
This case behave different between when reported and now.

Now(Current master):

### First format:

**Input:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: `   hello   `
```

**Output:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: ` hello `

```

### Second Format

**Input:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: ` hello `

```

**Output:**
```markdown
preserve leading and trailing no-break whitespaces in inline code: `hello`

```

**Comment by thorn0** [comment]
Fixed by #13590 in `next`
